### PR TITLE
feat(setup-wp-env): init

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: .github/actions/setup-wp-env/action.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    themes:
+        description: WordPress themes
+        required: false
+        default: ''
+    active-theme:
+        description: Active theme
+        required: false
+        default: ''
+    mappings:
+        description: Mappings directories into the WordPress container (e.g. templates:/path/to/local/templates)
```
Comment (StyleShit) [inline .github/actions/setup-wp-env/action.yml:24]:
```suggestion
        description: Mappings directories into the WordPress container (e.g. `templates:/path/to/local/templates`)
```

File: .github/actions/setup-wp-env/action.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+inputs:
+    wp:
+        description: WordPress version
+        required: true
+    php:
+        description: PHP version
+        required: true
+    plugins:
+        description: WordPress plugins
```
Comment (StyleShit) [inline .github/actions/setup-wp-env/action.yml:12]:
```suggestion
        description: A comma separated list of plugins to install (e.g. `elementor, elementor-pro`)
```

also others?
should it be paths? names?

File: .github/actions/setup-wp-env/action.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+name: Setup WP Env
+description: Setup WordPress environment for testing
+
+inputs:
+    wp:
+        description: WordPress version
+        required: true
+    php:
+        description: PHP version
+        required: true
```
Comment (StyleShit) [inline .github/actions/setup-wp-env/action.yml:10]: don't you allow empty and then it'll use `null`?

File: .github/actions/setup-wp-env/action.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+name: Setup WP Env
+description: Setup WordPress environment for testing
+
+inputs:
+    wp:
+        description: WordPress version
+        required: true
```
Comment (StyleShit) [inline .github/actions/setup-wp-env/action.yml:7]: same as above

File: .github/actions/setup-wp-env/action.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        description: PHP version
+        required: true
+    plugins:
+        description: WordPress plugins
+        required: false
+        default: ''
+    themes:
+        description: WordPress themes
+        required: false
+        default: ''
```
Comment (StyleShit) [inline .github/actions/setup-wp-env/action.yml:18]: same as the plugins

File: src/setup-wp-env/create-wp-env-config.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		dir,
+		'active-theme': activeTheme,
+	} = getOptions({
+		wp: { type: 'string', default: null },
+		php: { type: 'string', default: null },
+		plugins: { type: 'string', default: '' },
+		themes: { type: 'string', default: '' },
+		'active-theme': { type: 'string', default: '' },
+		mappings: { type: 'string', default: '' },
+		dir: { type: 'string', default: './' },
```
Comment (StyleShit) [inline src/setup-wp-env/create-wp-env-config.ts:19]: maybe rename dir to something more informative?

File: src/setup-wp-env/create-wp-env-config.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+function normalizeMappings(mappings: string) {
+	const config = normalizeArray(mappings)
+		.map((mapping) => mapping.split(':'))
+		.filter(([from, to]) => from && to);
+
+	return Object.fromEntries(config);
+}
+
+function normalizeArray(array: string) {
```
Comment (StyleShit) [inline src/setup-wp-env/create-wp-env-config.ts:72]: based on the name I would've expected an array parameter
maybe change to something like `parseAsArray` or something? IDK

File: src/setup-wp-env/create-wp-env-config.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+		boolean: entries
+			.filter(([, { type }]) => type === 'boolean')
+			.map(([key]) => key),
+		default: Object.fromEntries(
+			entries.map(([key, { default: value }]) => [key, value]),
+		),
+	});
+}
+
+function normalizeMappings(mappings: string) {
```
Comment (StyleShit) [inline src/setup-wp-env/create-wp-env-config.ts:64]:
```suggestion
function mappingsFromString(mappings: string) {
```

maybe?
and maybe do something similar for the array I said before?

File: tsup.config.ts [hunk 9]
```diff
@@ -1,13 +1,13 @@
 import { defineConfig } from 'tsup';
-import removeZxImportsEsbuildPlugin from './scripts/remove-zx-imports-esbuild-plugin.js';
+import removeZxImportsTsupPlugin from './scripts/remove-zx-imports-tsup-plugin.js';
```
Comment (StyleShit) [inline tsup.config.ts:2]:
```suggestion
import removeZxImportsTsupPlugin from './scripts/remove-zx-imports-tsup-plugin';
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: .github/actions/setup-wp-env/action.yml (hunk 1 above)
Comment (Nevoss) [reply .github/actions/setup-wp-env/action.yml:24]: Done

File: .github/actions/setup-wp-env/action.yml (hunk 2 above)
Comment (Nevoss) [reply .github/actions/setup-wp-env/action.yml:12]: Done

File: .github/actions/setup-wp-env/action.yml (hunk 3 above)
Comment (Nevoss) [reply .github/actions/setup-wp-env/action.yml:10]: Done

File: .github/actions/setup-wp-env/action.yml (hunk 4 above)
Comment (Nevoss) [reply .github/actions/setup-wp-env/action.yml:7]: Done

File: .github/actions/setup-wp-env/action.yml (hunk 5 above)
Comment (Nevoss) [reply .github/actions/setup-wp-env/action.yml:18]: Done

File: src/setup-wp-env/create-wp-env-config.ts (hunk 6 above)
Comment (Nevoss) [reply src/setup-wp-env/create-wp-env-config.ts:19]: Done

File: src/setup-wp-env/create-wp-env-config.ts (hunk 7 above)
Comment (Nevoss) [reply src/setup-wp-env/create-wp-env-config.ts:72]: Done

File: src/setup-wp-env/create-wp-env-config.ts (hunk 8 above)
Comment (Nevoss) [reply src/setup-wp-env/create-wp-env-config.ts:64]: Done

File: tsup.config.ts (hunk 9 above)
Comment (Nevoss) [reply tsup.config.ts:2]: Must have ext

**Review by StyleShit — APPROVED** [review]

**Pull request merged**